R/writedata.R

```diff
@@ -1,5 +1,92 @@
+#' the bare essential list entries, which can then be saved and called
+#' the same as an unreduced emulator by generate.TP.resids and
+#' generate.TP.fullgrids
+#'
+#' Note that with this reduced emulator, there is NO way to reconstruct
+#' the training data. A fully trained emulator contains a copy of the
+#' training data, in addition to the training regressor values (tgav),
+#' and the estimated linear model parameters and residuals
+#'  (meanfieldT$b, w, r), which together can also reconstruct the data.
+#'
+#' Even though the coordinate information stored in an emulator$griddataT
+#' is not needed directly to generate a new field of residuals or full data,
+#' it is often needed in downstream use of the fields. Therfore an entry
+#' reducedEmulator$griddataT$coord containg a matrix is saved in the
+#' reducedEmulator. Each is a matrix of coordinates for each grid cell, with
+#' cells in rows and latitude, longitude in the two columns. Keeping these
+#' coordinate matrices for T and P is negligible.
+#'
+#' Finally, the reduced emulator produced by this function is specifically
+#' meant for temperature and precipitation only, and is not robust to
+#' extension to other variables.
+#'
+#' Finally finally, if a user is interested in a different subset of
+#' list entries in a trained emulator, they are encouraged to subset and
+#' save themself, as appropriate for their project.
+#'
+#' @param emulator A trained fldgen emulator, with all entries needed
+#' for generating new residuals and for rigourously validating the
+#' quality of the trained emulator
+#'
+#' @return reducedEmulator A trained fldgen emulator with only the list
+#' entries needed by generate.TP.resids and generate.TP.fullgrids for
+#' generating new fields
+#'
+#' @author ACS July 2020
+#' @export
+emulator_reducer <- function(emulator){
+
+    if(length(names(emulator)) < 10){
+        stop('Your emulator is already reduced (missing at least one list entry)')
+    }
+
+    # There _is_ some weirdness to this structure, but it is intentional.
+    # We designed generate.TP.resids and generate.TP.fullgrids to just take
+    # a trained emulator because it was cleaner for the user. That means we need
+    # emulator$griddataP$pvarconvert_fcn to still exist so that calls in
+    # generate.TP.fullgrids still behave. But we can strip out everything
+    # else in emulator$griddataP.
+    # The alternative was to write  generate.TP.fullgrids
+    # with arguments that expect a user to provide
+    # p_convert = emulator$griddataP$pvarconvert_fcn from a stored emulator
+    # or to give p_convert = log. This basically opens up the option for
+    # users to make more errors.
+    # If anything, I'm inclined to remove the arguments
+    #  tvarunconvert_fcn = NULL, pvarunconvert_fcn = exp
+    # from generate.TP.fullgrids. We initially included things that way
+    # so that a user could experiment with variables other than T and P.
```

Comment:
>  The alternative was to write  generate.TP.fullgrids
    # with arguments that expect a user to provide
    # p_convert = emulator$griddataP$pvarconvert_fcn from a stored emulator
    # or to give p_convert = log. This basically opens up the option for
    # users to make more errors.
    # If anything, I'm inclined to remove the arguments
    #  tvarunconvert_fcn = NULL, pvarunconvert_fcn = exp
    # from generate.TP.fullgrids. We initially included things that way
    # so that a user could experiment with variables other than T and P.

That might be more helpful as an issue that as an inline comment.